features/topology-map/plugins/org.opennms.features.topology.plugins.topo.linkd/src/main/java/org/opennms/features/topology/plugins/topo/linkd/internal/TopologyFactory.java

```diff
@@ -0,0 +1,232 @@
+import org.opennms.features.topology.plugins.topo.linkd.internal.providers.CdpTopologyProvider;
+import org.opennms.features.topology.plugins.topo.linkd.internal.providers.EnhancedLinkdTopologyProvider;
+import org.opennms.features.topology.plugins.topo.linkd.internal.providers.IsisTopologyProvider;
+import org.opennms.features.topology.plugins.topo.linkd.internal.providers.LldpTopologyProvider;
+import org.opennms.features.topology.plugins.topo.linkd.internal.providers.OspfTopologyProvider;
+import org.opennms.netmgt.dao.api.BridgeBridgeLinkDao;
+import org.opennms.netmgt.dao.api.BridgeMacLinkDao;
+import org.opennms.netmgt.dao.api.BridgeTopologyDao;
+import org.opennms.netmgt.dao.api.CdpElementDao;
+import org.opennms.netmgt.dao.api.CdpLinkDao;
+import org.opennms.netmgt.dao.api.IpInterfaceDao;
+import org.opennms.netmgt.dao.api.IpNetToMediaDao;
+import org.opennms.netmgt.dao.api.IsIsLinkDao;
+import org.opennms.netmgt.dao.api.LldpElementDao;
+import org.opennms.netmgt.dao.api.LldpLinkDao;
+import org.opennms.netmgt.dao.api.NodeDao;
+import org.opennms.netmgt.dao.api.OspfLinkDao;
+import org.opennms.netmgt.dao.api.SnmpInterfaceDao;
+import org.opennms.netmgt.dao.api.TopologyDao;
+import org.opennms.netmgt.model.FilterManager;
+import org.springframework.transaction.support.TransactionOperations;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.collect.Lists;
+
+public class TopologyFactory {
+    
+    private MetricRegistry metricRegistry;
+    private NodeDao nodeDao;
+    private SnmpInterfaceDao snmpInterfaceDao;
+    private IpInterfaceDao ipInterfaceDao;
+    private TopologyDao topologyDao;
+    private FilterManager filterManager;
+    private boolean addNodeWithoutLink;
+    private LinkdHopCriteriaFactory linkdHopCriteriaFactory;
+    private LldpElementDao lldpElementDao;
+    private LldpLinkDao lldpLinkDao;
+    private CdpLinkDao cdpLinkDao;
+    private CdpElementDao cdpElementDao;
+    private OspfLinkDao ospfLinkDao;
+    private IsIsLinkDao isisLinkDao;
+    private BridgeMacLinkDao bridgeMacLinkDao;
+    private BridgeBridgeLinkDao bridgeBridgeLinkDao;
+    private BridgeTopologyDao bridgeTopologyDao;
+    private IpNetToMediaDao ipNetToMediaDao;
+    private TransactionOperations transactionOperations;
+    
+    public GraphProvider createBridgeTopologyProvider() {
+        BridgeTopologyProvider bridgeTopologyProvider = new BridgeTopologyProvider(metricRegistry);
+        injectEnhancedLinkdProperties(bridgeTopologyProvider);
+        bridgeTopologyProvider.setIpNetToMediaDao(ipNetToMediaDao);
+        bridgeTopologyProvider.setBridgeBridgeLinkDao(bridgeBridgeLinkDao);
+        bridgeTopologyProvider.setBridgeMacLinkDao(bridgeMacLinkDao);
+        bridgeTopologyProvider.setBridgeTopologyDao(bridgeTopologyDao);
+        bridgeTopologyProvider.setTopologyProviderInfo(new DefaultTopologyProviderInfo("Bridge", "Bridge"));
```

Comment:
The TopologyProviderInfo should contain a Name and Description. This seems to be a dummy description and name.